apps/web/pages/api/documents/index.ts

```diff
@@ -5,6 +5,8 @@ import { getUserFromToken } from "@documenso/lib/server";
 import prisma from "@documenso/prisma";
 import formidable from "formidable";
 
+const regex = /’/;
+
```

Comment:
How about we generate a random filename, instead of replacing characters. We can use UUID to generate unique file names. Example here: https://codesandbox.io/s/l5hnv

Reply:
My apologies I missed that this was about smart quotes, can we confirm that files with normal quotes don't break storing the file. If so we can just use the normal `encodeURIComponent` method before storage so we can still `decodeURIComponent` on the way out for the user while retaining safety on our end.

Reply:
- @piyushkrmaurya the files are not stored as files but in the database. The files with potentially tricky names come from user upload and are used to set the title of the document, so we cant choose the filename
- @Mythie yes, just checked. Works, just not the smart quotes. So you mean encode when saving and decode when reading? Makes sense, it would catch anything.
![image](https://github.com/documenso/documenso/assets/1309312/95c6564c-dd01-4b8d-9af5-aa7cf6e891bd)

Reply:
That's correct, we can percent encode the filename which will help us deal with most of these issues without needing to do anything drastic like slugify the filename :smile: